course/models.py

```diff
@@ -189,13 +190,23 @@ def namegen():
 post_save.connect(pseudonymize, sender=Enrollment)
 
 
+class UserTagManager(models.Manager): 
+    def set(self, **kwargs):
+        return self.get_or_create(**kwargs)
+
+    def unset(self, **kwargs):
+        self.filter(**kwargs).delete()
```

Comment:
Maybe `get` and `delete` would probably be safer here. Would ensure that only single object is found. Though, not-found needs to be handled.

Also, both should require `course_instance`.

Reply:
`get_or_create()` and `get()` behave similarly when multiple objects are found. They raise an exception: [`Queryset.get`](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#django.db.models.query.QuerySet.get)

These should have been renamed, I'll do that.
I propose renaming set, leaving it otherwise as it is now, and removing unset.
`unset()` is never used.
`unset()` is not used in taggings either, even though it is defined in the manager:

[definition](https://github.com/JoonatanPartanen/a-plus/blob/master/course/models.py#L235)

[`create()` is overridden, but `delete()` is not](https://github.com/JoonatanPartanen/a-plus/blob/master/course/api/full_serializers.py#L164)